app/notebook/server/NotebookManager.scala

```diff
@@ -54,7 +54,9 @@ class NotebookManager(val name: String, val notebookDir: File) {
     val sep = if (path.last == '/') "" else "/"
     val fpath = name.map(path + sep + _ + extension).getOrElse(incrementFileName(path + sep + "Untitled"))
     val nb = Notebook(
-      Some(new Metadata(getName(fpath),
+      Some(Metadata(
```

Comment:
Is there a case where we have no metadata? I don't think so in Enterprise, but I'm not sure in community. Maybe we can take this opportunity to make metadata a non-optional element?